microstream/build.gradle.kts

```diff
@@ -8,12 +8,13 @@ dependencies {
     api(libs.managed.microstream.storage.embedded.configuration)
 
     compileOnly(mn.micronaut.management)
-    implementation(libs.projectreactor)
+    implementation(mn.reactor)
```

Comment:
There is something curious in the fact that you have to import `micronaut-reactor`, but you are actually using `mn.reactor` and not `mnReactor.reactor`, as if `reactor` is already defined in `micronaut-core` (in which case there's no need to import the reactor catalog).

Reply:
There's no `reactor-test` in the core bom, there's just reactor core:

https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/gradle/libs.versions.toml#L124

Should we add `reactor-test` to core?  Or should we remove it from core, and rely on the version from `micronaut-reactor`?

Also, micronaut-reactor doesn't have managed dependencies any more, it has a bom version

Reply:
Any idea how we can add them

This contribution https://github.com/micronaut-projects/micronaut-reactor/pull/248 switched to using the reactor BOM instead

Reply:
When we generate a version catalog, we only inline definitions from other version catalogs, and reactor doesn't publish such a catalog. Therefore, you should add dependencies directly in the `reactor-bom` project of `micronaut-reactor`:

```gradle
catalog {
    versionCatalog {
        version("reactor", "3.5.0")
        library("reactor-core", "io.projectreactor", "reactor-core").versionRef("reactor")
    }
}
```

This isn't ideal, since we'd have to duplicate the version defined in the `pom.xml` file. However as you can see in [the reactor BOM](https://repo1.maven.org/maven2/io/projectreactor/reactor-bom/2022.0.0/reactor-bom-2022.0.0.pom), it doesn't define a common dependency version for several modules, so we would have to tweak it anyway.